flow/Arena.h

```diff
@@ -699,15 +698,18 @@ inline bool operator != (const StringRef& lhs, const StringRef& rhs ) { return !
 inline bool operator <= ( const StringRef& lhs, const StringRef& rhs ) { return !(lhs>rhs); }
 inline bool operator >= ( const StringRef& lhs, const StringRef& rhs ) { return !(lhs<rhs); }
 
-// This trait is used by VectorRef to determine if it should just memcpy the vector contents.
-// FIXME:  VectorRef really should use std::is_trivially_copyable for this BUT that is not implemented
-// in gcc c++0x so instead we will use this custom trait which defaults to std::is_trivial, which
-// handles most situations but others will have to be specialized.
+// This trait is used by VectorRef to determine if deep copy constructor should recursively
+// call deep copies of each element.
+// TODO: There should be an easier way to identify the difference between
+// flow_ref and non-flow_ref types.
 template <typename T>
-struct memcpy_able : std::is_trivial<T> {};
+struct non_flow_ref : std::is_fundamental<T> {};
 
```

Comment:
Why not use std::is_trivially_copyable? It's been supported since c++11, and we require that now. (Also the name non_flow_ref is confusing to me. There are things that are neither flow_refs nor trivially copyable).

Then, you wouldn't need the UID type (assuming UID is trivially copyable).

Is the issue that flow_ref is trivially copyable? (Maybe add a comment to this effect?)

Reply:
Yeah, the problem is that some flow ref types are trivially copyable. Also UID is not trivially copyable. I don't think there is a single standard type trait that can differentiate between the flow_ref types and the non-flow_ref types. I pushed a new commit that updates the comment to explain that `std::is_trivially_copyable` won't work.